cmd/chain/genesis/cmd.go

```diff
@@ -23,16 +24,27 @@ func Command() *cobra.Command {
 
 func genesisFunc(c *cobra.Command, args []string) error {
 	flags := c.Flags()
-	g, err := ParseFlags(flags, args)
+	config, err := ParseFlags(flags, args)
 	if err != nil {
 		return err
 	}
 
-	genesisBytes, err := genesis.Codec.Marshal(genesis.Version, g)
+	genesisBytes, err := genesis.Codec.Marshal(genesis.Version, config.Genesis)
 	if err != nil {
 		return err
 	}
 
-	_, err = os.Stdout.Write(genesisBytes)
+	if config.Encoding == binaryEncoding {
+		_, err = os.Stdout.Write(genesisBytes)
+		return err
+	}
+
+	// hex encoded
+	encoded, err := formatting.Encode(formatting.Hex, genesisBytes)
+	if err != nil {
+		return err
+	}
+	_, err = os.Stdout.WriteString(encoded)
+
```

Comment:
I feel like this should be formatted better. The reason the previous code used `os.Stdout` directly was because I wanted it to be able to be easily piped... I feel like this is more intended for a user to copy-paste... So I feel like `fmt.Println` would be better (to insert the newline at the end)... Thoughts?

Reply:
changed it